Assets/Plugins/Editor/Uplift/UpliftPreferences.cs

```diff
@@ -26,98 +26,115 @@
-        {
-            EnsurePrefsLoaded();
-            EditorGUILayout.LabelField("SSL Certificates:", EditorStyles.boldLabel);
-            EditorGUILayout.HelpBox("Uplift uses SSL certificates when updating itself, as it fetches its update from Github with HTTPS", MessageType.Info);
-            EditorGUILayout.HelpBox(
-                "Unknown certificates could be the result of our registered certificates being outdated or the result of a potential attack. Trusting unknown certificates could lead to security breaches. Use at your own risk!",
-                MessageType.Warning
-            );
-            trustUnknownCertificates = EditorGUILayout.Toggle("Trust unknown certificates", trustUnknownCertificates);
-
-            EditorGUILayout.LabelField("Experimental features:", EditorStyles.boldLabel);
-            EditorGUILayout.HelpBox(
-                "Experimental features are not thoroughly tested and could induce bugs. Use at your own risk!",
-                MessageType.Warning
-            );
-            useExperimentalFeatures = EditorGUILayout.Toggle("Use experimental features", useExperimentalFeatures);
-
-            EditorGUILayout.LabelField("Github related:", EditorStyles.boldLabel);
-            EditorGUILayout.HelpBox(
-                "Github does not support TLS versions 1.0 and 1.1 as of 22/02/2018, and TLS 1.2 is available in Unity only with Mono 4.8. If you want to use a Github Repository, it will need to be behind a proxy",
-                MessageType.Info
-            );
-            useGithubProxy = EditorGUILayout.Toggle("Use Github proxy", useGithubProxy);
-            GUI.enabled = useGithubProxy;
-            if (useGithubProxy)
-                EditorGUILayout.HelpBox(
-                    "The proxy URL will replace 'https://api.github.com'",
-                    MessageType.Info
-                );
-            githubProxyUrl = EditorGUILayout.TextField("Proxy URL", githubProxyUrl);
-            githubProxyUrl = Regex.Replace(githubProxyUrl, "/$", "");
-            GUI.enabled = true;
-
-            if (GUI.changed)
-            {
-                EditorPrefs.SetBool(useExperimentalFeaturesKey, useExperimentalFeatures);
-                EditorPrefs.SetBool(trustUnknownCertificatesKey, trustUnknownCertificates);
-                EditorPrefs.SetBool(githubProxyUseKey, useGithubProxy);
-                EditorPrefs.SetString(githubProxyUrlKey, githubProxyUrl);
-            }
-        }
+        private const string useExperimentalFeaturesKey = "UpliftExperimentalFeatures";
+        private const string trustUnknownCertificatesKey = "UpliftUnknownCertificates";
+        private const string githubProxyUseKey = "UpliftGithubProxyUseKey";
+        private const string githubProxyUrlKey = "UpliftGithubProxyUrlKey";
 
-        private static void EnsurePrefsLoaded()
+        private static UpliftSettings settingsCached = null;
+        private static UpliftSettings GetSettings(bool refresh = false)
```

Comment:
We do not have a place to force the reload. Should I put a menu item under Debug?

Reply:
I am more concerned by the fact we cache the results. E.g. what happens when we use the UI to edit it? Is the cache updated?

Reply:
Since the Uplift preferences follow the singleton pattern (which is bad, but the scope of this PR is not to refactor that) the cache should be updated. But anyhow, it shouldn't matter too much because settings/preferences are not likely to change at all during Unity runtime, so in my mind, a cache is very useful here to cut down a lot reading/serializing times.